pkg/auth/authorize-interceptor.go

```diff
@@ -0,0 +1,69 @@
+package auth
+
+import (
+	"context"
+	"log/slog"
+	"time"
+
+	"connectrpc.com/connect"
+	"github.com/metal-stack/api/go/request"
+	"github.com/metal-stack/metal-apiserver/pkg/repository"
+	"github.com/metal-stack/metal-apiserver/pkg/token"
+	"github.com/metal-stack/metal-lib/pkg/cache"
+)
+
+type (
+	authorizeInterceptor struct {
+		log        *slog.Logger
+		authorizer request.Authorizer
+	}
+)
+
+func NewAuthorizeInterceptor(log *slog.Logger, repo *repository.Store) *authorizeInterceptor {
+	// FIXME decide what is a proper cache timeout
+	projectAndTenantCache := cache.New(10*time.Millisecond, func(ctx context.Context, id string) (*request.ProjectsAndTenants, error) {
```

Comment:
Maybe we should not even add a cache as long as we do not see performance bottlenecks with this approach. In the OPA auther there was just a function for this, no cache.

Reply:
The actual performance penalty might be because we fetch the tenants and projects twice, first in the authorization, second in the tenant interceptor. We might think of storing the result of projectandtenantgetter in the context ?